### PR TITLE
Replace content of flang-compiler/f18 with "Moved" message

### PR DESCRIPTION
**Pull request by CarolineConcatto**
This patch replaces the content of flang-compiler-/f18 with a README.md file,as requested in task/issue[1].

The README  includes instructions for migrating commits from local fork to LLVM
and its new repository location.

Important: This patch will only be merged after the f18 was ported to LLVM
monorepo according to task/issue[2].

[1]https://github.com/flang-compiler/f18/issues/908
[2]https://github.com/flang-compiler/f18/issues/876

Signed-off-by: Caroline Concatto <caroline.concatto@arm.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,244 +1,36 @@
...
-  
-   Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-   See https://llvm.org/LICENSE.txt for license information.
-   SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
-  
--->
+# 301 - Moved
 
-# F18
+F18 is now part of LLVM.
```
Comment (kiranchandramohan) [inline README.md]: Should you also say 
-> that is available in the flang directory
-> renamed and available as flang in LLVM.

Comment (CarolineConcatto) [reply README.md]: I believe I should only say something like this:

Comment (DavidTruby) [reply README.md]: This wording works for me

Comment (CarolineConcatto) [reply README.md]: Done!

**Comment by sscalpone** [comment]
How did you test the migration processes that are outlined in the readme?

**Comment by DavidTruby** [comment]
> How did you test the migration processes that are outlined in the readme?

I tested it using the flattened version of the repo I've been using to get ready for upstreaming. Once the CMake patch is merged I can re-flatten and push that out to my fork of llvm-project along with Eric's LLVM-side cmake changes. That should make it easy for people to test both the commands here and building F18/Flang in-tree with LLVM, as well as see what the flattened commit history will look like.

**Comment by sscalpone** [comment]
How about reducing this PR to just the message?  It will be easier to review & keep correct.  Then, after the upstream move, we can do another PR to remove all of the files.

**Comment by RichBarton-Arm** [comment]
I think there is a significant advantage in the commit history having the addition of the readme and the removal of the files as an atomic commit.

I don't think it will be hard to co-ordinate. If we can agree on the contents of the readme beforehand, David can update this PR after we push into LLVM and then someone with F18 commit access can merge it before commit access is shut off. There should be no risk of any changes taking place on the F18 side after we have merged into LLVM.

**Comment by DavidTruby** [comment]
We also don't want the changed README going in to LLVM, so we have to do it all after we merge anyway

**Comment by sscalpone** [comment]
My point is that we can (easily) approve the text of the message now whereas we cannot audit the list of files to be deleted until after we've moved things to LLVM.  My preference is to push README.md as soon as the code is moved to LLVM so there's no delay while we collect the list of files to be deleted.

With that, I'll defer to @CarolineConcatto on how to proceed.

**Comment by CarolineConcatto** [comment]
@sscalpone either way is fine.
 The solution to having only one PR is good for git history. As it will be only one PR for deleting and letting a message.
 If you are worried about the files, @DavidTruby and I could work together to guarantee that all the files that went in LLVM were also in the F18 project. 
Moreover, we could still review the Readme and later see the files, and only merge once we are happy about both. Not a problem for me.
But you have the power to do the merge. So you have the word.

**Comment by sscalpone** [comment]
It is ok if the PR combines the changes to the README and the deletion of all files.  We can finish the review after the PR after f18 is merged to llvm.org.  The text in the README is ok now.

**Pull request merged**